tls/ttls.c

```diff
@@ -1064,17 +1066,14 @@ ttls_hdr_check(TlsCtx *tls)
 	}
 	/* Check length against bounds of the current transform and version */
 	if (!ttls_xfrm_ready(tls)) {
+		/* Cipher's not ready yet, plaintext limits apply. */
 		if (io->msglen < 1 || io->msglen > TLS_MAX_PAYLOAD_SIZE) {
 			T_DBG("bad message length %u\n", io->msglen);
 			return T_DROP;
 		}
 	} else {
-		/*
-		 * TLS encrypted messages can have up to 256 bytes of padding.
-		 */
-		if (io->msglen < tls->xfrm.minlen
-		    || io->msglen
-		       > tls->xfrm.minlen + TLS_MAX_PAYLOAD_SIZE + 256)
+		if (io->msglen < tls->xfrm.minlen ||
+		    io->msglen > ttls_max_ciphertext_len(&tls->xfrm))
```

Comment:
`256` inherited from mbedTLS - is it wrong? If not, then we need to describe this in comment for `ttls_max_ciphertext_len()`.

Reply:
> is it wrong

It's not wrong, it's just too forgiving. We are not supporting CBC modes, only AEAD ciphers, and those are not using padding. So there is no need to keep that additional 256 bytes in the size limit. I've added a comment to the ttls_max_ciphertext_len().